packages/react-core/src/components/Menu/MenuItem.tsx

```diff
@@ -290,49 +298,66 @@ const MenuItemBase: React.FunctionComponent<MenuItemProps> = ({
       role="none"
       {...props}
     >
-      <Component
-        tabIndex={-1}
-        className={css(styles.menuItem, getIsSelected() && styles.modifiers.selected, className)}
-        aria-current={getAriaCurrent()}
-        disabled={isDisabled}
-        role="menuitem"
-        ref={innerRef}
-        onClick={(event: any) => {
-          onItemSelect(event, onSelect);
-          _drill && _drill();
-        }}
-        {...additionalProps}
-      >
-        <span className={css(styles.menuItemMain)}>
-          {direction === 'up' && (
-            <span className={css(styles.menuItemToggleIcon)}>
-              <AngleLeftIcon aria-hidden />
-            </span>
-          )}
-          {icon && <span className={css(styles.menuItemIcon)}>{icon}</span>}
-          <span className={css(styles.menuItemText)}>{children}</span>
-          {isExternalLink && (
-            <span className={css(styles.menuItemExternalIcon)}>
-              <ExternalLinkAltIcon aria-hidden />
-            </span>
-          )}
-          {(flyoutMenu || direction === 'down') && (
-            <span className={css(styles.menuItemToggleIcon)}>
-              <AngleRightIcon aria-hidden />
-            </span>
-          )}
-          {getIsSelected() && (
-            <span className={css(styles.menuItemSelectIcon)}>
-              <CheckIcon aria-hidden />
+      <GenerateId>
+        {randomId => (
+          <Component
+            tabIndex={-1}
+            className={css(styles.menuItem, getIsSelected() && !hasCheck && styles.modifiers.selected, className)}
+            aria-current={getAriaCurrent()}
+            {...(!hasCheck && { disabled: isDisabled })}
+            role="menuitem"
```

Comment:
From the core docs, if the menu item is a checkbox item (`<label>`), this item is set to `role="none"` and the parent `.pf-c-menu__list-item` element gets `role="menuitem"`